tardis/tests/tests_slow/conftest.py

```diff
@@ -39,6 +42,53 @@ def pytest_unconfigure(config):
     shutil.rmtree(config.option.tempdir)
 
 
+@pytest.mark.hookwrapper
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    report = outcome.get_result()
+    plot_object = getattr(item, "plot_object", None)
+    report.extra = plot_object.plot_links
+
+
+@pytest.fixture(scope="function")
+def plot_object(request):
+    class PlotUploader(object):
+        def __init__(self):
+            self._plots = list()
+            self.plot_links = list()
+
+        def add(self):
+            # TODO: make this accept pyplot.figure later.
+            self._plots = glob.glob(os.path.join(
+                request.config.option.tempdir, "w7", "*.png"))
+
+        def upload(self):
+            dokuwiki_url = request.config.dokureport.dokuwiki_url
+            for plot in self._plots:
+                request.config.dokureport.doku_conn.medias.add("plots:{0}_{1}".format(
+                    tardis.__githash__[0:7], plot.split("/")[-1]), plot
+                )
+                self.plot_links.append(extras.url(
+                    "{0}lib/exe/fetch.php?media=plots:{1}_{2}".format(
+                        dokuwiki_url, tardis.__githash__[0:7], plot.split("/")[-1]
+                    )
+                ))
+                os.unlink(plot)
+
+    plot_obj = PlotUploader()
+    setattr(request.node, "plot_object", plot_object)
+
+    def fin():
+        # Adding plots in this manner is temporary.
+        plot_obj.add()
```

Comment:
Yes it is temporary. It will be changed when `pyplot.figures` are directly passed on.